segmentstore/server/src/test/java/io/pravega/segmentstore/server/logs/ThrottlerTests.java

```diff
@@ -145,6 +161,191 @@ public void testInterruptedDurableDataLogDelay() throws Exception {
+        // currently running throttle cycle and request the next throttling value.
+        for (int i = 1; i < suppliedDelays.size(); i++) {
+            // Interrupt the current throttle cycle.
+            t.notifyThrottleSourceChanged();
+            Assert.assertFalse("Not expected throttle future to be completed yet.", t1.isDone());
+
+            // Wait for the new cycle to begin (we use the recordDelay consumer above to figure this out).
+            int expectedDelayCount = i + 1;
+            TestUtils.await(() -> delays.size() == expectedDelayCount, 5, TIMEOUT_MILLIS);
+        }
+        TestUtils.await(t1::isDone, 5, TIMEOUT_MILLIS);
+
+        String[] tags = { "container", String.valueOf(this.containerId), "throttler", "Cache" };
+
+        // Because the supplied delays is monotonically decreasing, only the first delay value should be used to calculate
+        // the duration supplied.
+        AssertExtensions.assertLessThanOrEqual(
+                "Throttler should be at most the first supplied delay",
+                (int) suppliedDelays.get(0),
+                (int) MetricRegistryUtils.getGauge(MetricsNames.OPERATION_PROCESSOR_DELAY_MILLIS, tags).value()
+        );
+    }
+
+    /**
+     * Tests if interruptible throttlers are correctly reporting the time spent throttled.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testInterruptedDecreasingDelayMetrics() throws Exception {
+        // Supply monotonically decreasing delays.
+        val suppliedDelays = Arrays.asList(5000, 4000, 3000);
+        val delays = Collections.<Integer>synchronizedList(new ArrayList<>());
+        val calculator = new TestCalculatorThrottler(ThrottlerCalculator.ThrottlerName.Cache);
+        val nextDelay = suppliedDelays.iterator();
+        Consumer<Integer> recordDelay = delayMillis -> {
+            delays.add(delayMillis);
+            calculator.setDelayMillis(nextDelay.hasNext() ? nextDelay.next() : 0); // 0 means we're done (no more throttling).
+        };
+        @Cleanup
+        TestThrottler t = new TestThrottler(this.containerId, wrap(calculator), executorService(), metrics, recordDelay);
+
+        // Set a non-maximum delay and ask to throttle, then verify we throttled the correct amount.
+        calculator.setDelayMillis(nextDelay.next());
+        val t1 = t.throttle();
+        Assert.assertFalse("Not expected throttle future to be completed yet.", t1.isDone());
+
+        // For every delay that we want to submit, notify that the cache cleanup has completed, which should cancel the
+        // currently running throttle cycle and request the next throttling value.
+        for (int i = 1; i < suppliedDelays.size(); i++) {
+            // Interrupt the current throttle cycle.
+            t.notifyThrottleSourceChanged();
+            //Assert.assertFalse("Not expected throttle future to be completed yet.", t1.isDone());
```

Comment:
Please do not check in commented out code. Either add it back or delete it completely.